From e7d8e4fb09e053dd624df0b7c78c1f8ea41c9ed7 Mon Sep 17 00:00:00 2001
From: Nic Costa <nic.costa@gmail.com>
Date: Mon, 6 Apr 2020 15:49:37 -0500
Subject: [PATCH] fix calculation of remaining buffer size when calling update
 scripts

the buffer used to hold the full shell command to call the upgrade
scripts during FOTA is of length ARM_UC_MAXIMUM_COMMAND_LENGTH
(currently 256), but the caluclation for space remaining after printing
a file path to the buffer was incorrectly using a different value,
ARM_UC_MAXIMUM_FILE_AND_PATH_LENGTH (currently 128) as the buffer
length, causing the function to fail incorrectly.

the fix is to use the buffer length passed into the function rather
than using a different hardcoded #define.
---
 .../source/arm_uc_pal_linux_implementation_internal.c  | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/lib/mbed-cloud-client/update-client-hub/modules/pal-linux/source/arm_uc_pal_linux_implementation_internal.c b/lib/mbed-cloud-client/update-client-hub/modules/pal-linux/source/arm_uc_pal_linux_implementation_internal.c
index f4ae0f2..4e54d50 100755
--- a/lib/mbed-cloud-client/update-client-hub/modules/pal-linux/source/arm_uc_pal_linux_implementation_internal.c
+++ b/lib/mbed-cloud-client/update-client-hub/modules/pal-linux/source/arm_uc_pal_linux_implementation_internal.c
@@ -175,7 +175,7 @@ static bool arm_uc_pal_linux_internal_command(arm_ucp_worker_t *parameters,
             }
 
             /* update remaining */
-            remaining = ARM_UC_MAXIMUM_FILE_AND_PATH_LENGTH - length;
+            remaining = command_length - length;
 
             /* check validity */
             valid = ((result.error == ERR_NONE) && (remaining > 0));
@@ -203,7 +203,7 @@ static bool arm_uc_pal_linux_internal_command(arm_ucp_worker_t *parameters,
             }
 
             /* update remaining */
-            remaining = ARM_UC_MAXIMUM_FILE_AND_PATH_LENGTH - length;
+            remaining = command_length - length;
 
             /* check validity */
             valid = ((result.error == ERR_NONE) && (remaining > 0));
@@ -218,7 +218,7 @@ static bool arm_uc_pal_linux_internal_command(arm_ucp_worker_t *parameters,
                                *arm_uc_location);
 
             /* update remaining */
-            remaining = ARM_UC_MAXIMUM_FILE_AND_PATH_LENGTH - length;
+            remaining = command_length - length;
 
             /* check validity */
             valid = (remaining > 0);
@@ -233,7 +233,7 @@ static bool arm_uc_pal_linux_internal_command(arm_ucp_worker_t *parameters,
                                arm_uc_offset);
 
             /* update remaining */
-            remaining = ARM_UC_MAXIMUM_FILE_AND_PATH_LENGTH - length;
+            remaining = command_length - length;
 
             /* check validity */
             valid = (remaining > 0);
@@ -249,7 +249,7 @@ static bool arm_uc_pal_linux_internal_command(arm_ucp_worker_t *parameters,
                                    arm_uc_buffer->size_max);
 
                 /* update remaining */
-                remaining = ARM_UC_MAXIMUM_FILE_AND_PATH_LENGTH - length;
+                remaining = command_length - length;
 
                 /* check validity */
                 valid = (remaining > 0);
-- 
2.21.0 (Apple Git-122)

